### PR TITLE
Camel jira

### PR DESCRIPTION
**Pull request by ffang**

### CODE REVIEW

**Comment by ffang** [comment]
native build can make some progress, but I still see error like
`com.oracle.svm.core.util.UserError$UserException: No instances are allowed in the image heap for a class that is initialized or reinitialized at image runtime: sun.security.provider.NativePRNG. Try marking this class for build-time initialization with --initialize-at-build-time=sun.security.provider.NativePRNG
Detailed message:
Trace: 	object java.security.SecureRandom
	method com.google.api.client.auth.oauth.OAuthParameters.computeNonce()
Call path from entry point to com.google.api.client.auth.oauth.OAuthParameters.computeNonce(): 
	at com.google.api.client.auth.oauth.OAuthParameters.computeNonce(OAuthParameters.java:120)
	at org.apache.camel.component.jira.oauth.JiraOAuthAuthenticationHandler.configure(JiraOAuthAuthenticationHandler.java:62)
`

If add --initialize-at-build-time=sun.security.provider.NativePRNG as the error message indicate, I got another error like
`
Error: Incompatible change of initialization policy for sun.security.provider.NativePRNG: trying to change BUILD_TIME from the command line to RERUN for substitutions
com.oracle.svm.core.util.UserError$UserException: Incompatible change of initialization policy for sun.security.provider.NativePRNG: trying to change BUILD_TIME from the command line to RERUN for substitutions
`

File: extensions/jira/deployment/pom.xml [hunk 1]
```diff
@@ -0,0 +1,57 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (WillemJiang) [inline extensions/jira/deployment/pom.xml]: Apache License header.

File: integration-tests/jira/src/main/java/org/apache/camel/quarkus/component/jira/it/JiraResource.java [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+    @Inject
+    ConsumerTemplate consumerTemplate;
+
+    @Path("/get")
+    @GET
+    @Produces(MediaType.TEXT_PLAIN)
+    public String get() throws Exception {
+        final String message = consumerTemplate.receiveBodyNoWait("jira:newIssues?jiraUrl=" + JIRA_CREDENTIALS, String.class);
+        log.infof("Received from jira: %s", message);
+        return "message";
```
Comment (WillemJiang) [inline integration-tests/jira/src/main/java/org/apache/camel/quarkus/component/jira/it/JiraResource.java]: Why don't you send the message back?

**Comment by zhfeng** [comment]
@ffang [Quarks native tips](https://quarkus.io/guides/writing-native-applications-tips#delay-class-initialization) could be helpful with the failures.

File: extensions/jira/deployment/pom.xml (hunk 1 above)
Comment (ffang) [reply extensions/jira/deployment/pom.xml]: @WillemJiang , Apache License header added, thanks!

File: integration-tests/jira/src/main/java/org/apache/camel/quarkus/component/jira/it/JiraResource.java (hunk 2 above)
Comment (ffang) [reply integration-tests/jira/src/main/java/org/apache/camel/quarkus/component/jira/it/JiraResource.java]: send the message back, but it's just empty string

**Comment by ffang** [comment]
@zhfeng , thanks for the info, I will take a close look.

**Comment by lburgazzoli** [comment]
@ffang have you had any time to progress with this pr ?

**Comment by ffang** [comment]
@lburgazzoli, not yet, I hope to have some time to revisit it

**Comment by ffang** [comment]
I will close this PR since this is messy and out-of-date, will send a new PR for camel-jira quarkus extension with native mode fixed

**Pull request closed without merging**